llmfoundry/callbacks/hf_checkpointer.py

```diff
@@ -39,6 +40,12 @@ class HuggingFaceCheckpointer(Callback):
         huggingface_folder_name (str): Folder to save each checkpoint under (can be a format string). Default is ``ba{batch}``.
         precision: The precision to save the model in. Default is ``float32``. Options are ``bfloat16``, ``float16``, or ``float32``.
         overwrite (bool): Whether to overwrite previous checkpoints.
+        log_to_mlflow (bool): Whether to log and register the checkpoint to MLFlow. Default is ``False``.
+        mlflow_task (str): The MLFlow task to log the checkpoint under. Only used if ``log_to_mlflow`` is ``True``. Default is ``text-generation``.
+        mlflow_metadata (Optional[dict]): The MLFlow metadata to log the checkpoint with. Only used if ``log_to_mlflow`` is ``True``. Default is ``None``.
+        uc_prefix: (Optional[str]): Prefix to use for the MLFlow registered model. If specified, the model will be logged to UC rather than
+            the workspace model registry. If specified, the prefix must be of the form ``{catalog}.{schema}``
+            The model will be registered at ``{catalog}.{schema}.{model name}``. Only used if ``log_to_mlflow`` is ``True``. Default is ``None``.
```

Comment:
I think you should make these arguments to the MLflowLogger not to this callback? Seems like these fall under configuring how you log to mlflow not configuring your huggingface checkpoint. 
Pros: less mlflow set up logic in this callback and less for anyone else that wants to log stuff to mlflow

Reply:
hmm, i sort of agree...however, the task and metadata are tied to the model type....

Reply:
ideally this would all be in your "inference checkpointer" callback in composer :)

Reply:
also these args are _specific_ to the transformers model flavor